crowbar_framework/app/models/node_object.rb

```diff
@@ -256,6 +256,32 @@ def intended_role=(value)
     @node["crowbar_wall"]["intended_role"] = value
   end
 
+  def disk_roles=(value)
+    @node["crowbar_wall"] ||= {}
+    @node["crowbar_wall"]["disk_roles"] = value
+  end
+
+  def disk_roles
+    disk_roles = crowbar_wall["disk_roles"] || {}
+
+    # remove devices that were claimed after they role was set
+    disk_roles.each do |d, _role|
+      crowbar_wall["disk_roles"].delete d if crowbar_wall[:claimed_disks].key? d
+    end
```

Comment:
Instead of using `disk_roles.each` + `delete`, it should be possible to use `reject` on the `crowbar_wall` directly, I think. Same below on line 277. More generally, is the `disk_roles` helper variable needed at all?